### ISSUE TITLE
Integration fails to set up after update/restart

### ISSUE THREAD

**Issue by Benniepie** [kvj/hass_Bluetooth_Proxy#1]
Hello,

Thank you for building this integration and app.

I had set it set up successfully and it appeared to be working. After an update/restart of HA the integration then fails to set up. Attempts to reload/restarts haven't worked so thought I should report it here.

HA 2024.1.0b5
App 0.1.1

Log as follows:
```
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 406, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/companion_bt_proxy/__init__.py", line 39, in async_setup_entry
    scanner = CompanionBLEScanner(hass, entry)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/companion_bt_proxy/scanner.py", line 11, in __init__
    super().__init__(hass, entry.entry_id, entry.title, bluetooth.async_get_advertisement_callback(hass), None, False)
  File "src/habluetooth/base_scanner.py", line 194, in habluetooth.base_scanner.BaseHaRemoteScanner.__init__
TypeError: __init__() takes exactly 4 positional arguments (6 given)
```

**Comment by kvj** [comment]
There was indeed a number of breaking changes (which weren't considered breaking by their authors). I've updated the integration to 0.1.1 which should hopefully restore the functionality

**Comment by Benniepie** [comment]
Thank you, your latest version has resolved the error - I appreciate the fast response and fix. I've got quite a few android phones with broken screens or in use as touch screen controls that I will be ideal as bluetooth proxies. so will be adding a few more to fix my flakey HA bluetooth devices.

I had a feeling the issue may have been the bluetooth code changes,  there has been a lot of BT code changes lately and I noticed a conscious decision not to publicise the breaking changes as they assumed no bluetooth proxies had been developed as custom components outside of HA core. 

Thanks again for writing it!
